data/provider/src/main/java/tech/pegasys/teku/api/ValidatorDataProvider.java

```diff
@@ -141,6 +165,34 @@ private SafeFuture<UInt256> retrieveExecutionPayloadValue(final UInt64 slot) {
         .orElseThrow(() -> new IllegalStateException("Execution Payload Value is not available"));
   }
 
+  private SafeFuture<UInt256> retrieveConsensusBlockRewards(final BlockContainer blockContainer) {
+    return combinedChainDataClient
+        .getStateAtSlotExact(blockContainer.getBlock().getSlot().decrement())
+        .thenApply(
+            maybeParentState ->
+                maybeParentState.map(
+                    parentState ->
+                        rewardCalculator.getBlockRewardData(blockContainer, parentState)))
+        .thenApply(
+            blockRewardData ->
+                blockRewardData.map(
+                    rewardData -> EthConstants.GWEI_TO_WEI.multiply(rewardData.getTotal())))
+        .thenApply(
+            maybeTotalRewards -> {
+              if (maybeTotalRewards.isEmpty()) {
+                LOG.warn("Unable to calculate block rewards. Setting value to 0");
+                return UInt256.ZERO;
+              } else {
+                return maybeTotalRewards.get();
+              }
+            })
+        .exceptionally(
+            throwable -> {
+              LOG.error("Unable to calculate block value", throwable);
+              return null;
```

Comment:
If we are in PHASE0, we will get here, see https://github.com/Consensys/teku/blob/2a3fd9fbfa1253ad1bd4bcc864774b622758779f/data/provider/src/main/java/tech/pegasys/teku/api/RewardCalculator.java#L78-L83, so this block will completes with null as consensus value, which could broke something next (not sure but maybe). So regarding this comment https://github.com/Consensys/teku/pull/7574#discussion_r1376078234, I'd add the same thing here
```
LOG.warn("Unable to calculate block rewards. Setting value to 0", ex);
return UInt256.ZERO;
```

Reply:
Refactored it to return 0 in that case and logs a warning. Added a test too